compiler/ballerina-lang/src/main/java/io/ballerina/compiler/api/impl/symbols/BallerinaUnionTypeSymbol.java

```diff
@@ -147,6 +154,32 @@ public String signature() {
         return this.signature;
     }
 
+    @Override
+    public boolean isEnum() {
+        if (this.enumSymbol != null) {
+            return true;
+        }
+
+        return this.getBType().tsymbol.getKind() == SymbolKind.ENUM;
+    }
+
+    @Override
+    public Optional<EnumSymbol> getEnumSymbol() {
+        if (this.enumSymbol != null) {
+            return Optional.of(this.enumSymbol);
+        }
+
+        BTypeSymbol tsymbol = this.getBType().tsymbol;
+        if (tsymbol.getKind() != SymbolKind.ENUM) {
+            return Optional.empty();
+        }
+
+        SymbolFactory symbolFactory = SymbolFactory.getInstance(this.context);
+        this.enumSymbol = symbolFactory.createEnumSymbol((BEnumSymbol) tsymbol, tsymbol.getName().value);;
+
+        return Optional.ofNullable(this.enumSymbol);
```

Comment:
```suggestion
        return Optional.of(this.enumSymbol);
```
At this point can the enumSymbol be null?

Reply:
Thanks for noticing!